### DISCUSSION TITLE
How to set the upper limit of resources by tenants with "Database-Oriented Multi-Tenancy" setup?

### DISCUSSION THREAD

**Discussion by jubingc** [milvus-io/milvus#33660] [Q&A and General discussion]
Hi,

I have a multi-tenancy setup where we host a shared cluster for our users. Each user has their own account and database, with access limited to their data only. Essentially, each user/tenant manages their own database, collections, and partitions.

I am wondering if there is a way to limit the cluster resources (CPU/memory/storage/network) used by each database/tenant/user. My goal is to ensure that the cluster resources are efficiently utilized across all tenants while guaranteeing that each tenant has sufficient resources and does not interfere with others.

Is there a way to achieve this resource management within the shared cluster?

Thank you!

**Comment by yhmo** [comment]
This tool helps to estimate resources required according to your total data size:
https://milvus.io/tools/sizing

Resource is always for the total data size, not for a single tenant.
